AutoRest/Generators/CSharp/CSharp/Templates/MethodTemplate.cshtml

```diff
@@ -52,9 +62,9 @@ public async Task<@(Model.OperationResponseReturnTypeString)> @(Model.Name)WithH
     {
         invocationId = ServiceClientTracing.NextInvocationId.ToString();
         Dictionary<string, object> tracingParameters = new Dictionary<string, object>();
-    @foreach (var parameter in Model.LocalParameters)
+    @foreach (var parameter in Model.ParameterTemplateModels.Where(p => !p.IsParameterGroup && p.ClientProperty == null))
```

Comment:
@devigned @stankovski 
Please do a quick review. With this change a client model method's parameters list gets a little complicated that it no long simply maps to the argument list to emit out for the method declarations, so more condition checks were introduced to filter based on the context. 
This works, but if you have better suggestion, please comment here before it gets merged.
For "ungrouping", we probably won't do it because other than storage team's ask, there is less benefits; rather the author can tweak the spec easily.